### ISSUE TITLE
[FEATURE] - Add more flexibility and functionality to the MERGE statement

### ISSUE THREAD

**Issue by AmineSagaama** [ebiznext/comet-data-pipeline#388]
## Description
Add the possibility to merge a part of a BigQuery Table, rather than all the Table.
We should, add a filter in MERGE metadata options, in order to apply the MERGE, only on a part of the table.

For example, if we want, to merge, only the last partition in the Table, we should read only this parititon,
and Apply the MERGE statement.

## Solution proposition
I suggest to add a query option in the Merge Metadata case class, if the query is specified, we should apply the merge,
only on this part of the table.

```
case class MergeOptions(
  key: List[String],
  delete: Option[String] = None,
  timestamp: Option[String] = None,
  query: Option[String]
)
```

## Additional context
We should update the Excel and Yaml schema, to take account of this feature.

## Complexity estimation
T-Shirt Size: M

**Comment by hayssams** [comment]
Could you provide an example of the query please

**Comment by AmineSagaama** [comment]
Actually, when we apply the MERGE statement, we read the entire table : 

```
val bqTable = s"${domain.name}.${schema.name}"
                val existingBigQueryDF = session.read
                  // We provided the acceptedDF schema here since BQ lose the required / nullable information of the schema
                  .schema(withScriptFieldsDF.schema)
                  .format("com.google.cloud.spark.bigquery")
                  .option("table", bqTable)
                  .load()
                merge(withScriptFieldsDF, existingBigQueryDF, mergeOptions)
```

In case of a REINIT of a specific partition in the Table (partition of January for example) , I want to read only the specific partition from the Table, and apply the merge :

```
val existingBigQueryDF = session.read
                  // We provided the acceptedDF schema here since BQ lose the required / nullable information of the schema
                  .schema(withScriptFieldsDF.schema)
                  .format("com.google.cloud.spark.bigquery")
                  .option("table", bqTable)
                   .option("filter", " event_date = '30-01-2020' ")
                  .load()
                merge(withScriptFieldsDF, existingBigQueryDF, mergeOptions)
```
In case of a Snapshot Table, I only want to read the last partition, and apply the merge : 

```
val existingBigQueryDF = session.read
                  // We provided the acceptedDF schema here since BQ lose the required / nullable information of the schema
                  .schema(withScriptFieldsDF.schema)
                  .format("com.google.cloud.spark.bigquery")
                  .option("table", bqTable)
                   .option("filter", " event_date = 'date of the last snapshot in the BigQuery Table' ")
                  .load()
                merge(withScriptFieldsDF, existingBigQueryDF, mergeOptions)
```

**Comment by hayssams** [comment]
gr8 explanation.
Let's call the field "query-filter" instead of query.

**Comment by AmineSagaama** [comment]
solved by #397
